### ISSUE TITLE
Hard Stuttering

### ISSUE THREAD

**Issue by OutsideOctaves** [slyh/Cities2-TrafficLightsEnhancement#57]
**Describe the bug**
I have a city that is around 100K CIMs, and without this mod it runs as expected.  A few moments after launching the game and loading the map it begins a "hard stutter"  That is that the simulation and game goes into a hard stutter that is not based in the FPS of the video output but in the sim I believe, but if I reload it without the mod, the stutter goes away.  I tried it on a new city and eventually a few minutes in it began again.

**To Reproduce**
No real idea here.
I'd say just load a large map with a ton of traffic lights using the advanced option and ticking the exclusive ped. faze.

**Expected behaviour**
I expect to be able to load the map with mod enabled and run it as it was before this last update.  Previous versions ran just fine.  (Though that does make me sound entitled, I'm not... but if you ask me the expected behavior lol)
**Screenshots**
N/A

**Log file**
Done, and dusted.

**Additional context**
N/A

**Comment by slyh** [comment]
Could you provide a log file or tell me which version exhibits this issue, as well as the previous version that did not have this problem?

**Comment by OutsideOctaves** [comment]
Log file should have uploaded with it.  hmmm let me try again here 
[LogOutput.log](https://github.com/slyh/Cities2-TrafficLightsEnhancement/files/13656372/LogOutput.log)

**Comment by slyh** [comment]
At first, I thought it might be due to the switch to React, but version 0.2.3 hadn't incorporated that change yet.

Which version of this mod are you using that doesn't exhibit this issue? I have a city with 130k cims and can maintain a simulation speed of 0.6-0.8x most of the time, with occasional drops to 0.2x. (CPU: Ryzen 3900X, mod version: v0.2.7-alpha, game version: 1.0.18f1)

The performance might have improved with the 1.0.18f1 update. If you're still experiencing this problem, could you send me your save file so that I can try to replicate it?

**Comment by OutsideOctaves** [comment]
Updated today to 18f1 and before I updated TFE, I tried just with the version I had which was the current highest version on thunderstore... and it was smooth again.  Something in the update fixed the issue I was having... so far.  Then I updated everything on thunderstore and it is still smooth going.  I'm guessing the version I was on was 2.3?  if that was the one we had before 2.4?  (I'm on the alpha version).  It's now 2.7.

**Comment by slyh** [comment]
The only version available on Thunderstore was 0.2.3, but it has been updated to 0.2.7 today. The major difference is that the newer version includes the rewritten UI, which may be slower but is easier to maintain.

Feel free to reopen this issue if it regresses.
